metrics-forwarder/internal/client/client.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	datadog "github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+)
+
+type apiClient interface {
+	CallAPI(req *http.Request) (*http.Response, error)
+	PrepareRequest(ctx context.Context, path string, method string, postBody interface{}, headerParams map[string]string, queryParams url.Values, formParams url.Values, fileName *datadog.FormFile) (*http.Request, error)
+}
+
+type DatadogClient struct {
```

Comment:
not immediate but this part is a re-usable code when creating log forwarding client as well